flent/runners.py

```diff
@@ -845,6 +846,95 @@ def find_binary(cls, host, interval, length, ip_version, local_bind=None,
         raise RuntimeError("No suitable Iperf binary found.")
 
 
+class ParseException(Exception):
+    pass
+
+
+class SsRunner(ProcessRunner):
+    """Runner for iterated `ss -t -i -p`. Depends on same partitial output
+    separationa and time stamping as TcRunner."""
+
+    ip_v4_addr_sub_re = "(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])"
+    ip_v6_addr_sub_re = "(?:(?:[A-F0-9]{1,4}:){7}[A-F0-9]{1,4}|(?=(?:[A-F0-9]{0,4}:){0,7}[A-F0-9]{0,4}(?![:.\w]))(([0-9A-F]{1,4}:){1,7}|:)((:[0-9A-F]{1,4}){1,7}|:))(:\d+)+"
```

Comment:
While I can appreciate the preciseness of these, I fear they are going to be hard to debug going forward. I wonder if it can be simplified; for instance having something like "([0-9]{1,3}\.){3}[0-9]{1,3}" match an IPv4 address would be fine; the risk of having addresses >255 show up is not that high.

But, well, on the other hand I can live with those if you prefer them. But in that case, please at least comment the different parts of the IPv6 regex so it is readable to those with less regex foo (I'm not sure I get the different parts myself of that one).